src/optionsdialog.cpp

```diff
@@ -179,6 +179,9 @@ void OptionsDialog::writeSettings(void)
     settings->endGroup();
     ///-----PROGRAMING--LANGUAGE---------------------------------------------------
     settings->beginGroup("compilation_settings");
+    settings->setValue("defaultLanguage/text",defaultLanguageComboBox->currentText());
```

Comment:
Do you really need to store default language combobox text? Looks like you use only index...